src/processors/modulation/Panner.cpp

```diff
@@ -304,7 +304,13 @@ bool Panner::getCustomComponents (OwnedArray<Component>& customComps)
             mainPanSlider.setVisible (! dualPanOn);
             leftPanSlider.setVisible (dualPanOn);
 
-            setName (vts.getParameter (dualPanOn ? leftPanTag : mainPanTag)->name);
+            auto newName = vts.getParameter (dualPanOn ? leftPanTag : mainPanTag)->name;
+            if(newName != getName())
+            {
+                setName(newName);
+                if(auto* parent = getParentComponent())
+                    getParentComponent()->repaint();
+            }
```

Comment:
I'm a bit confused on what the `timerCallback` does here. From what I can see whenever the parameter is changed `parameterChanged()` is called, so when would the timerCallback be necessary? Is there an edge case I'm missing.

Reply:
Hmm, so for PanSlider2, I remember that we needed the `timerCallback()` in order to update the UI depending on if the module was receiving a stereo or mono input. It's possible that the `timerCallback()` in this class is just a leftover from that, since it looks like the call to `updateSliderVisiblity()` is made redundant by the parameter listener. Have you tried turning off the timer and seeing if anything changes?

Reply:
Yeah I haven't seen any changes when getting rid of the timer and doing some testing, I'll take it out and merge!